src/plugins/vis_type_table/public/components/table_vis_component.tsx

```diff
@@ -20,10 +21,38 @@ interface TableVisComponentProps {
 }
 
 export const TableVisComponent = ({ table, visConfig, handlers }: TableVisComponentProps) => {
-  const { rows, columns } = table;
+  const { formattedRows: rows, formattedColumns: columns } = convertToFormattedData(
+    table,
+    visConfig
+  );
 
   const pagination = usePagination(visConfig, rows.length);
 
+  // set sort and resize column state
+  const [sortColumn, setSortColumn] = useState<SortColumn>(
+    handlers.uiState.get('vis.sortColumn') || {}
+  );
+
+  const [columnsWidth, setColumnsWidth] = useState<ColumnWidth[]>(
+    handlers.uiState.get('vis.columnsWidth') || []
+  );
+
+  // store current state
+  const curColumnsState = useRef<{
+    sortColumn: SortColumn;
+    columnsWidth: ColumnWidth[];
+  }>({
+    sortColumn: handlers.uiState.get('vis.sortColumn'),
+    columnsWidth: handlers.uiState?.get('vis.columnsWidth'),
```

Comment:
nit - if you actually need `handlers.uiState?.` here, you may need it elsewhere - I'd expect it to be consistent.

Reply:
will fix this.